tagging/tag_service.py

```diff
@@ -134,32 +134,83 @@ def add_tags(self, tags: List[Tag], asset_uid: str) -> Dataset:
         asset_tags_uid : str
             ID of the existing tag set to update
 
-
         Returns
         ----------
-        Dataset
-            The updated tag set (full asset tag object)
+        added_tags_uid : List[str]
+            List of the UIDs of the added tags
         """
+        added_tags_uid = []
+        # If no tags were sent, returns empty list
+        if tags is None:
+            return added_tags_uid
 
-        doc_tags = self._collection_dataset.find_one({'uid': asset_uid})
-        if not doc_tags:
+        asset = self._collection_dataset.find_one({'uid': asset_uid})
+        if not asset:
             raise DatasetNotFound(f"no asset with id: {asset_uid}")
 
-        if not doc_tags.get('tags'):
-            doc_tags['tags'] = []
+        if asset['tags'] is None:
+            self._collection_dataset.update_one(
+                {'uid': asset_uid},
+                {'$set': {'tags': []}})
 
-        for tag in tags:
-            doc_tags['tags'].append(tag.dict())
+        tags_dict = []
+        for user_tag in tags:
+            tag = Tag(**user_tag.dict())
```

Comment:
I don't think this translation of UserTag to Tag back to dict is necessary.

Pydantic has a nice [utility](https://stackoverflow.com/questions/55762673/how-to-parse-list-of-models-with-pydantic) function that is useful here.

Reply:
I made the following change:

    persisted_tags = parse_obj_as(List[PersistedTag], tags)
    added_tags_uid = list(map(attrgetter('uid'), persisted_tags)) 
    persisted_tags_dict = [persisted_tag.dict() for persisted_tag in persisted_tags]
    self._collection_dataset.update_one(
            {'uid': dataset_uid},
            {'$push': {'tags': {'$each': persisted_tags_dict}}}
        )

I am still creating the dictionary because persisted_tags is a List[PersistedTag], so I do not think I can send it directly to mongo.

Let me know what you think about this change.

Reply:
Sorry, my comment was about the code for building the return objects, after the mongo update:
```
        tags_dict = []
        for user_tag in tags:
            tag = Tag(**user_tag.dict())
```

Reply:
I see. The mongo update before these lines is to initialize the tags as an empty list in mongo (I think by default mongo fills this field as Null instead of []), and the ones after that appends the elements in the list of tags.

I just sent the updated version. Let me know if there are further changes to make.